adala/runtimes/_openai.py

```diff
@@ -1,224 +1,46 @@
-
-    Returns:
-        Dict[str, Any]: OpenAI response or error message.
-    """
-    openai_api_key = openai_api_key or os.getenv("OPENAI_API_KEY")
-    messages = [{"role": "user", "content": user_prompt}]
-    if system_prompt:
-        if instruction_first:
-            messages.insert(0, {"role": "system", "content": system_prompt})
-        else:
-            messages[0]["content"] += system_prompt
-
-    try:
-        completion = await client.chat.completions.create(
-            model=model,
-            messages=messages,
-            max_tokens=max_tokens,
-            temperature=temperature,
-        )
-        completion_text = completion.choices[0].message.content
-        return {
-            "text": completion_text,
-            "_adala_error": False,
-            "_adala_message": None,
-            "_adala_details": None,
-        }
-    except Exception as e:
-        # Handle other exceptions
-        return {
-            "text": None,
-            "_adala_error": True,
-            "_adala_message": type(e).__name__,
-            "_adala_details": str(e),
-        }
-
+from adala.utils.parse import parse_template, partial_str_format
+from openai import NotFoundError, OpenAI
+from pydantic import ConfigDict, Field, computed_field
+from rich import print
 
-async def async_concurrent_create_completion(
-    prompts,
-    client,
-    instruction_first,
-    openai_model,
-    max_tokens,
-    temperature,
-):
-    tasks = [
-        asyncio.ensure_future(
-            async_create_completion(
-                client=client,
-                user_prompt=prompt["user"],
-                system_prompt=prompt["system"],
-                model=openai_model,
-                max_tokens=max_tokens,
-                temperature=temperature,
-                instruction_first=instruction_first,
-            )
-        )
-        for prompt in prompts
-    ]
-    responses = await asyncio.gather(*tasks)
-    return responses
+from .base import Runtime
+from ._litellm import (
+    LiteLLMChatRuntime,
+    AsyncLiteLLMChatRuntime,
+    LiteLLMVisionRuntime,
+)
 
 
-class OpenAIChatRuntime(Runtime):
+class OpenAIChatRuntime(LiteLLMChatRuntime):
     """
     Runtime that uses [OpenAI API](https://openai.com/) and chat completion models to perform the skill.
 
     Attributes:
-        openai_model: OpenAI model name.
+        model: OpenAI model name.
         openai_api_key: OpenAI API key. If not provided, will be taken from OPENAI_API_KEY environment variable.
         base_url: Can point to any implementation of the OpenAI API. Defaults to OpenAI's.
         max_tokens: Maximum number of tokens to generate. Defaults to 1000.
     """
 
-    model_config = ConfigDict(arbitrary_types_allowed=True)  # for @computed_field
-
-    openai_model: str = Field(alias="model")
-    openai_api_key: Optional[str] = Field(
-        default=os.getenv("OPENAI_API_KEY"), alias="api_key"
+    model: str
```

Comment:
ah, yeah this is somewhat questionable. We're renaming this field here (bc I want it to override the litellm one, and it doesn't make sense to name it "openai_model" in litellm...)
We could add an alias (not a pydantic one, just a separate attr), but... idk it feels like an annoying thing to support, and seems unlikely anyone's using it.
Wdyt?

Reply:
Same goes for api_key, btw

Reply:
this is an acceptable level of breakage imo